src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -90,68 +88,45 @@ const StudentApplication = ({ userProfile, authentication }) => {
       setUnsavedChanges(true);
     };
 
-    if (!applicationID || !applicationDetails) {
-      setLoading(true);
-      // Check if the current user is on an application. Returns the application ID number if found
-      housing
-        .getCurrentApplicationID()
-        .then((newApplicationID) => {
-          console.log('Retrieved Application ID: ' + newApplicationID); //! DEBUG
-          if (newApplicationID === null || newApplicationID === -1) {
-            // Intentionally trigger the 'catch'
-            throw new Error("Invalid value of 'newApplicationID' = " + newApplicationID);
-          } else if (isSubscribed) {
-            setApplicationID(newApplicationID);
+    const loadApplication = async () => {
+      try {
+        setLoading(true);
+        // Check if the current user is on an application. Returns the application ID number if found
+        const newApplicationID = await housing.getCurrentApplicationID();
+        if (newApplicationID === null || newApplicationID === -1) {
+          // Intentionally trigger the 'catch'
+          throw new Error("Invalid value of 'newApplicationID' = " + newApplicationID);
```

Comment:
Maybe instead of using an error to trigger the catch, we should just call the `handleError` function inside this `if`, especially since there is nothing after the else anyways